packages/postgraphile-core/__tests__/kitchen-sink-schema.sql

```diff
@@ -1238,3 +1239,67 @@ create table geometry.geom (
   polygon polygon,
   circle circle
 );
+
+--------------------------------------------------------------------------------
+
+create schema function_returning_enum;
+
+create table function_returning_enum.stage_options (
+  type text primary key
+);
+comment on table function_returning_enum.stage_options is E'@enum';
+insert into function_returning_enum.stage_options (type) values ('pending'), ('round 1'), ('round 2'), ('rejected'), ('hired');
+
+create table function_returning_enum.applicants (
+  id int,
+  first_name text,
+  last_name text,
+  stage text references function_returning_enum.stage_options (type)
+);
+
+--- follow the convention of [enum_name]_enum_domain
+create domain function_returning_enum.stage_options_enum_domain as text;
+
+create or replace function function_returning_enum.applicants_next_stage(
```

Comment:
No need for `or replace` here, and it requires the reader to validate the function hasn't already been defined.

```suggestion
create function function_returning_enum.applicants_next_stage(
```